javalanglib/src/main/scala/java/lang/System.scala

```diff
@@ -207,11 +210,54 @@ object System {
     }
   }
 
-  //def getProperties(): java.util.Properties
-  //def getProperty(key: String): String
-  //def getProperty(key: String, default: String): String
-  //def clearProperty(key: String): String
-  //def setProperty(key: String, value: String): String
+  private object SystemProperties {
+    var value = loadSystemProperties()
+
+    private def loadSystemProperties(): ju.Properties = {
+      val sysProp = new ju.Properties()
+      sysProp.setProperty("java.version", "1.8")
+      sysProp.setProperty("java.vm.specification.version", "1.8")
+      sysProp.setProperty("java.vm.specification.vendor", "Oracle Corporation")
+      sysProp.setProperty("java.vm.specification.name", "Java Virtual Machine Specification")
+      sysProp.setProperty("java.vm.name", "Scala.js")
+      sysProp.setProperty("java.vm.version", linkerVersion)
+      sysProp.setProperty("java.specification.version", "1.8")
+      sysProp.setProperty("java.specification.vendor", "Oracle Corporation")
+      sysProp.setProperty("java.specification.name", "Java Platform API Specification")
+      sysProp.setProperty("file.separator", "/")
+      sysProp.setProperty("path.separator", ":")
+      sysProp.setProperty("line.separator", "\n")
+
+      for {
+        jsEnvProperties <- environmentInfo.javaSystemProperties
+        (key, value) <- jsEnvProperties
+      } {
+        sysProp.setProperty(key, value)
+      }
+      sysProp
+    }
+  }
+
+  def getProperties(): ju.Properties =
+    SystemProperties.value
+
+  def setProperties(properties: ju.Properties): Unit = {
+    if (properties == null)
+      throw new scala.NullPointerException
+    SystemProperties.value = properties
```

Comment:
JavaDoc says:
> If the argument is null, then the current set of system properties is forgotten.

Experimentation suggests that it does recreate a new set out of `loadSystemProperties()` (not an empty map). So it should probably be:
```scala
SystemProperties.value =
  if (properties != null) properties
  else SystemProperties.loadSystemProperties()
```